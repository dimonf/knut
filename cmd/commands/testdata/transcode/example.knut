include "USD.prices"
include "AAPL.prices"

* Open Accounts

2019-12-31 open Equity:Equity
2019-12-31 open Assets:BankAccount
2019-12-31 open Assets:Portfolio

2019-12-31 open Expenses:Groceries
2019-12-31 open Expenses:Fees
2019-12-31 open Expenses:Rent

2019-12-31 open Income:Salary
2019-12-31 open Income:Dividends

* Opening Balances

2019-12-31 "Opening balance"
Equity:Equity           Assets:BankAccount           10000 CHF

* 2020-01

2020-01-25 "Salary January 2020"
Income:Salary           Assets:BankAccount            5000 CHF

2020-01-02 "Rent January"
Assets:BankAccount      Expenses:Rent                 2000 CHF

2020-01-15 "Groceries"
Assets:BankAccount      Expenses:Groceries             200 CHF

2020-01-05 "Transfer to portfolio"
Assets:BankAccount      Assets:Portfolio              1000 CHF

2020-01-06 "Currency exchange"
Equity:Equity           Assets:Portfolio              1001 USD
Assets:Portfolio        Equity:Equity                  969 CHF

2020-01-06 "Buy 3 AAPL shares"
Equity:Equity           Assets:Portfolio                12 AAPL
Assets:Portfolio        Equity:Equity                  900 USD
Assets:Portfolio        Expenses:Fees                    4 USD

* 2020-02

2020-02-25 "Salary January 2020"
Income:Salary           Assets:BankAccount            5000 CHF

2020-02-02 "Rent January"
Assets:BankAccount      Expenses:Rent                 2000 CHF

2020-02-05 "Groceries"
Assets:BankAccount      Expenses:Groceries             250 CHF

2020-02-25 "Groceries"
Assets:BankAccount      Expenses:Groceries             423 CHF
