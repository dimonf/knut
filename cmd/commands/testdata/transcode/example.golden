option "operating_currency" "CHF"

2019-12-31 open Equity:Equity

2019-12-31 open Assets:BankAccount

2019-12-31 open Assets:Portfolio

2019-12-31 open Expenses:Groceries

2019-12-31 open Expenses:Fees

2019-12-31 open Expenses:Rent

2019-12-31 open Income:Salary

2019-12-31 open Income:Dividends

2019-12-31 * "Opening balance"
  Equity:Equity -10000 CHF
  Assets:BankAccount 10000 CHF

2020-01-02 * "Rent January"
  Assets:BankAccount -2000 CHF
  Expenses:Rent 2000 CHF

2020-01-05 * "Transfer to portfolio"
  Assets:BankAccount -1000 CHF
  Assets:Portfolio 1000 CHF

2020-01-06 * "Buy 3 AAPL shares"
  Equity:Equity -873.74907696 CHF
  Assets:Portfolio 873.74907696 CHF
  Assets:Portfolio -874.332 CHF
  Equity:Equity 874.332 CHF
  Assets:Portfolio -3.88592 CHF
  Expenses:Fees 3.88592 CHF

2020-01-06 * "Currency exchange"
  Equity:Equity -972.45148 CHF
  Assets:Portfolio 972.45148 CHF
  Assets:Portfolio -969 CHF
  Equity:Equity 969 CHF

2020-01-07 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -6.77688552 CHF
  Income:Portfolio 6.77688552 CHF

2020-01-07 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.28906 CHF
  Income:Portfolio 0.28906 CHF

2020-01-08 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -14.24652324 CHF
  Assets:Portfolio 14.24652324 CHF

2020-01-08 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.03201 CHF
  Assets:Portfolio 0.03201 CHF

2020-01-09 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -22.77704496 CHF
  Assets:Portfolio 22.77704496 CHF

2020-01-09 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.42389 CHF
  Assets:Portfolio 0.42389 CHF

2020-01-10 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -1.96919412 CHF
  Assets:Portfolio 1.96919412 CHF

2020-01-10 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.00776 CHF
  Income:Portfolio 0.00776 CHF

2020-01-13 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -19.37438604 CHF
  Assets:Portfolio 19.37438604 CHF

2020-01-13 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.00194 CHF
  Assets:Portfolio 0.00194 CHF

2020-01-14 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -14.7839352 CHF
  Income:Portfolio 14.7839352 CHF

2020-01-14 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.23668 CHF
  Income:Portfolio 0.23668 CHF

2020-01-15 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -7.29269484 CHF
  Income:Portfolio 7.29269484 CHF

2020-01-15 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.35211 CHF
  Income:Portfolio 0.35211 CHF

2020-01-15 * "Groceries"
  Assets:BankAccount -200 CHF
  Expenses:Groceries 200 CHF

2020-01-16 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -8.12763108 CHF
  Assets:Portfolio 8.12763108 CHF

2020-01-16 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.32689 CHF
  Income:Portfolio 0.32689 CHF

2020-01-17 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -11.21830104 CHF
  Assets:Portfolio 11.21830104 CHF

2020-01-17 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.11446 CHF
  Assets:Portfolio 0.11446 CHF

2020-01-20 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -3.18411276 CHF
  Assets:Portfolio 3.18411276 CHF

2020-01-20 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.32301 CHF
  Assets:Portfolio 0.32301 CHF

2020-01-21 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -6.11256168 CHF
  Income:Portfolio 6.11256168 CHF

2020-01-21 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.01649 CHF
  Assets:Portfolio 0.01649 CHF

2020-01-22 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -3.68312184 CHF
  Assets:Portfolio 3.68312184 CHF

2020-01-22 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.04074 CHF
  Assets:Portfolio 0.04074 CHF

2020-01-23 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -3.106026 CHF
  Assets:Portfolio 3.106026 CHF

2020-01-23 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.1358 CHF
  Income:Portfolio 0.1358 CHF

2020-01-24 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -0.66471744 CHF
  Income:Portfolio 0.66471744 CHF

2020-01-24 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.2037 CHF
  Assets:Portfolio 0.2037 CHF

2020-01-25 * "Salary January 2020"
  Income:Salary -5000 CHF
  Assets:BankAccount 5000 CHF

2020-01-27 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -26.81569956 CHF
  Income:Portfolio 26.81569956 CHF

2020-01-27 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.04268 CHF
  Assets:Portfolio 0.04268 CHF

2020-01-28 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -25.34602728 CHF
  Assets:Portfolio 25.34602728 CHF

2020-01-28 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.00873 CHF
  Income:Portfolio 0.00873 CHF

2020-01-29 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -22.39403676 CHF
  Assets:Portfolio 22.39403676 CHF

2020-01-29 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.30361 CHF
  Assets:Portfolio 0.30361 CHF

2020-01-30 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -1.1775798 CHF
  Income:Portfolio 1.1775798 CHF

2020-01-30 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.0194 CHF
  Assets:Portfolio 0.0194 CHF

2020-01-31 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -45.42511752 CHF
  Income:Portfolio 45.42511752 CHF

2020-01-31 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.36569 CHF
  Income:Portfolio 0.36569 CHF

2020-02-02 * "Rent January"
  Assets:BankAccount -2000 CHF
  Expenses:Rent 2000 CHF

2020-02-03 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -8.0741862 CHF
  Income:Portfolio 8.0741862 CHF

2020-02-03 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.58685 CHF
  Income:Portfolio 0.58685 CHF

2020-02-04 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -31.68825384 CHF
  Assets:Portfolio 31.68825384 CHF

2020-02-04 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.22698 CHF
  Assets:Portfolio 0.22698 CHF

2020-02-05 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -10.9752012 CHF
  Assets:Portfolio 10.9752012 CHF

2020-02-05 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.34629 CHF
  Assets:Portfolio 0.34629 CHF

2020-02-05 * "Groceries"
  Assets:BankAccount -250 CHF
  Expenses:Groceries 250 CHF

2020-02-06 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -14.86509624 CHF
  Assets:Portfolio 14.86509624 CHF

2020-02-06 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.39091 CHF
  Assets:Portfolio 0.39091 CHF

2020-02-07 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -13.9729506 CHF
  Income:Portfolio 13.9729506 CHF

2020-02-07 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.1164 CHF
  Assets:Portfolio 0.1164 CHF

2020-02-10 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -6.52732884 CHF
  Assets:Portfolio 6.52732884 CHF

2020-02-10 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.20952 CHF
  Assets:Portfolio 0.20952 CHF

2020-02-11 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -5.26228776 CHF
  Income:Portfolio 5.26228776 CHF

2020-02-11 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.04268 CHF
  Assets:Portfolio 0.04268 CHF

2020-02-12 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -20.77624896 CHF
  Assets:Portfolio 20.77624896 CHF

2020-02-12 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.1455 CHF
  Income:Portfolio 0.1455 CHF

2020-02-13 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -4.9092552 CHF
  Income:Portfolio 4.9092552 CHF

2020-02-13 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.19012 CHF
  Assets:Portfolio 0.19012 CHF

2020-02-14 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -1.52146332 CHF
  Assets:Portfolio 1.52146332 CHF

2020-02-14 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.12804 CHF
  Assets:Portfolio 0.12804 CHF

2020-02-17 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -2.73932868 CHF
  Assets:Portfolio 2.73932868 CHF

2020-02-17 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.27257 CHF
  Assets:Portfolio 0.27257 CHF

2020-02-18 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -18.75773184 CHF
  Income:Portfolio 18.75773184 CHF

2020-02-18 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.12513 CHF
  Income:Portfolio 0.12513 CHF

2020-02-19 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -16.01548212 CHF
  Assets:Portfolio 16.01548212 CHF

2020-02-19 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.2425 CHF
  Assets:Portfolio 0.2425 CHF

2020-02-20 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -9.21316764 CHF
  Income:Portfolio 9.21316764 CHF

2020-02-20 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.0582 CHF
  Assets:Portfolio 0.0582 CHF

2020-02-21 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -21.146946 CHF
  Income:Portfolio 21.146946 CHF

2020-02-21 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.02522 CHF
  Assets:Portfolio 0.02522 CHF

2020-02-24 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -48.2866584 CHF
  Income:Portfolio 48.2866584 CHF

2020-02-24 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.47724 CHF
  Income:Portfolio 0.47724 CHF

2020-02-25 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -28.84647816 CHF
  Income:Portfolio 28.84647816 CHF

2020-02-25 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.09118 CHF
  Assets:Portfolio 0.09118 CHF

2020-02-25 * "Groceries"
  Assets:BankAccount -423 CHF
  Expenses:Groceries 423 CHF

2020-02-25 * "Salary January 2020"
  Income:Salary -5000 CHF
  Assets:BankAccount 5000 CHF

2020-02-26 * "Adjust value of AAPL in account Assets:Portfolio"
  Income:Portfolio -10.02634956 CHF
  Assets:Portfolio 10.02634956 CHF

2020-02-26 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.37636 CHF
  Income:Portfolio 0.37636 CHF

2020-02-27 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -55.60483824 CHF
  Income:Portfolio 55.60483824 CHF

2020-02-27 * "Adjust value of USD in account Assets:Portfolio"
  Income:Portfolio -0.04753 CHF
  Assets:Portfolio 0.04753 CHF

2020-02-28 * "Adjust value of AAPL in account Assets:Portfolio"
  Assets:Portfolio -6.7340898 CHF
  Income:Portfolio 6.7340898 CHF

2020-02-28 * "Adjust value of USD in account Assets:Portfolio"
  Assets:Portfolio -0.74108 CHF
  Income:Portfolio 0.74108 CHF

